File: example/go/models/post.go
package models

import (
	"example/go/enums"
)

type Post struct {
	Id string `gorm:"primaryKey;column:id" json:"id"`
	Title *string `gorm:"column:title" json:"title"`
	Body *string `gorm:"column:body" json:"body"`
	Slug *string `gorm:"column:uuid" json:"slug"`
	Status enums.PostStatus `gorm:"column:status;not null;default:draft" json:"status"`
	UserId *string `gorm:"column:user_id" json:"userId"`
	CategoryId *string `gorm:"column:category_id" json:"categoryId"`
	Tags []Tag `gorm:"column:tags" json:"tags"`
}

File: example/lib/models/post/post.dart
import 'package:schemix/schemix.dart';

import '../category/category.dart';
import '../tag/tag.dart';
import '../user/user.dart';

part 'post.schemix.dart';

enum PostStatus { draft, published, archived }

@Schemix(
  name: 'posts',
  schemaVersion: 1,
  enableTimestamps: true,
  enableSoftDelete: true,
)
class Post {
  @PrimaryKey(autoGenerate: true)
  String id;

  @Length(min: 1, max: 255)
  String? title;

  @FullTextSearch()
  @SchemixField(searchable: true)
  String? body;

  @DatabaseGenerated(strategy: 'uuid')
  String? slug;

  @DatabaseDefault(PostStatus.draft)
  PostStatus status = PostStatus.draft;

  @BelongsTo(User)
  String? userId;

  @BelongsTo(Category)
  String? categoryId;

  @ManyToMany(Tag, junctionTable: 'post_tags')
  List<Tag> tags = [];

  Post({required this.id});

  factory Post.fromJson(Map<String, dynamic> json) => _$PostFromJson(json);
  Map<String, dynamic> toJson() => _$PostToJson(this);
  Post copy() => _$PostCopy(this);
}

// class Pop {
//   final post = Post(id: "");

//   void doSomething() {
//     Post.
//   }
// }
